src/core/annotation.js

```diff
@@ -59,25 +59,39 @@ class AnnotationFactory {
    * @param {Object} ref
    * @param {PDFManager} pdfManager
    * @param {Object} idFactory
+   * @param {boolean} checkJSActions
    * @returns {Promise} A promise that is resolved with an {Annotation}
    *   instance.
    */
-  static create(xref, ref, pdfManager, idFactory) {
-    return pdfManager.ensureCatalog("acroForm").then(acroForm => {
+  static create(xref, ref, pdfManager, idFactory, checkJSActions = true) {
```

Comment:
Can't we make `false` the default value instead, and explicitly enable it where necessary?